FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js

```diff
@@ -65,6 +67,67 @@ const GeneralInfo = (props) => {
     const { data: fetchedActivities, isLoading: isActivitiesLoading } = useSWR(`${process.env.REACT_APP_BASE_API_URL}/api/activities/`, fetcher);
     const { data: fetchedCategories, isLoading: isCategoriesLoading } = useSWR(`${process.env.REACT_APP_BASE_API_URL}/api/categories/`, fetcher);
 
+    const { setFormIsDirty } = useContext(DirtyFormContext);
+
+    // TODO: update default values as new fields added
+
+    const defaultValues = {
+        'name': mainProfile?.name,
+        'official_name': mainProfile?.official_name ?? null,
+        'edrpou': mainProfile?.edrpou ?? null,
+        'region': mainProfile?.region ?? null,
+        'categories': mainProfile?.categories ?? [],
+        'activities': mainProfile?.activities ?? [],
+        'banner_image': mainProfile?.banner_image ?? null,
+        'logo_image': mainProfile?.logo_image ?? null,
+        'common_info': mainProfile?.common_info ?? null,
+        'is_registered': mainProfile?.is_registered ?? '',
+        'is_startup': mainProfile?.is_startup ?? '',
+    };
+
+    const compareActivitisCategories = (array1, array2) => {
+        return (
+            array1.length === array2.length &&
+            array1.every((element_1) =>
+                array2.some((element_2) =>
+                    Object.keys(element_1).every((key) => element_1[key] === element_2[key])
+                )
+            )
+        );
+      };
+
+    // TODO: remove logic for 'edrpou' field check when it  will be replaced with char field on server side
+
+    const checkFormIsDirty = () => {
+        let isDirty = false;
+        Object.keys(defaultValues).forEach((key) => {
+          if (key === 'categories' || key === 'activities') {
+            if (!compareActivitisCategories(defaultValues[key], profile[key])) {
+                isDirty = true;
+                return;
+            }
+          } else if (key === 'edrpou') {
+                if (((defaultValues[key] && profile[key] !== null) && (defaultValues[key].toString() !== profile[key].toString())) ||
+                    (defaultValues[key] === null && profile[key] !== null)) {
+                    isDirty = true;
+                    return;
+                }
+          } else if (defaultValues[key] !== profile[key]) {
+                if (defaultValues[key] === null && profile[key] === '') {
+                    return;
+                }
+                isDirty = true;
+                return;
+          }
+        });
```

Comment:
I would recommend creating a shared util function that will make this check. When we need to make a specific check like for `categories` we may add `type` property to `defaultValues` for running the specific check.

Reply:
Done. checkFormIsDirty util function created.